clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7163,7 +7165,8 @@ void Sema::ProcessDeclAttributeList(
     } else if (const auto *A = D->getAttr<OpenCLIntelReqdSubGroupSizeAttr>()) {
       Diag(D->getLocation(), diag::err_opencl_kernel_attr) << A;
       D->setInvalidDecl();
-    } else if (!D->hasAttr<CUDAGlobalAttr>()) {
+    } else if (!D->hasAttr<CUDAGlobalAttr>() &&
```

Comment:
This part checks amdgpu kernel attributes. I think we should move it inside the above if body and change it to
```
if (!D->hasAttr<CUDAGlobalAttr>() && !FnTy->getCallConv() != CallingConv::CC_AMDGPUKernelCall) {
//...
}
```
instead of using `else if`.

Reply:
Updated, I think this is what you meant?